common/src/main/java/de/maxhenkel/voicechat/config/ServerConfig.java

```diff
@@ -33,17 +33,17 @@ public ServerConfig(ConfigBuilder builder) {
         voiceChatPort = builder
                 .integerEntry("port", 24454, -1, 65535,
                         "The port of the voice chat server",
-                        "Setting this to \"-1\" sets the port to the Minecraft servers port (Not recommended)"
+                        "Setting this to '-1' sets the voice chat port to the Minecraft server port (Not recommended)"
```

Comment:
I've noticed that people get confused by this and thing the voice chat will uses the games networking instead of UDP. We should add another line here, clarifying that it just use the same port number but still uses a separate UDP port.

Reply:
Here's my suggestion for what the entire new option description could look like:
> The port number to use for the voice chat communication on the server side.
> Audio packets are always transmitted via the UDP protocol on the port number
> specified here, independently of other networking used for the game server.
> Set this to '-1' to use the same port number that is used by the Minecraft server.

You can take this as a starting point for further modifications or just scrap it entirely and suggest something much better. This also shows that periods at the end of sentences (which you don't use in comments in configuration files at all, and I consistently stick to this rule) are really useful, as you can see in the multi-line example above.

Reply:
"on the server side" would imply that its server side only. 

>The port to use for the voice chat communication.
Audio packets are always transmitted via UDP on the port number
specified here, independently of other networking used for the game server.
Set this to '-1' to use the same port number that is used by the Minecraft server - This is not recommended since by default this port is already in use and will crash the server.

This is very long though...

Reply:
Will this crash the server? How is this possible?

As far as I know, regular client-server communication in Minecraft happens over TCP, typically on port 25565 by default. This mod, however (I'm only speculating here, as I haven't really delved into the code), in addition to the standard vanilla-based networking, also uses UDP for transmitting audio between players. So there shouldn't be any port conflicts, right?

---

As for the phrase "on the server side" that I added, it may indeed imply that the configured port number is a server-side-only thing. But isn't that true? The user can only configure the server port, while the client just gets assigned a random port from the pool and has no control over it.

A separate issue, however, is the so-called integrated server that is used when playing in the single-player mode. But in that case, there's no need to configure anything, and the voice chat just works. Or rather, it doesn't, since there's no one to hear you anyway unless you are also using a mod like Voice Chat Interaction, for example (but it is only available for Fabric, which I find a bit unfortunate and I hope that one day you will port it to (Neo)Forge as well :) ).

Reply:
> Will this crash the server? How is this possible?

Minecrafts "server-query" uses UDP and runs on port 25565

Reply:
> As for the phrase "on the server side" that I added, it may indeed imply that the configured port number is a server-side-only thing. But isn't that true? The user can only configure the server port, while the client just gets assigned a random port from the pool and has no control over it.

Thats correct, but still sounds weird

Reply:
> A separate issue, however, is the so-called integrated server that is used when playing in the single-player mode. But in that case, there's no need to configure anything, and the voice chat just works. Or rather, it doesn't, since there's no one to hear you anyway unless you are also using a mod like Voice Chat Interaction, for example (but it is only available for Fabric, which I find a bit unfortunate and I hope that one day you will port it to (Neo)Forge as well :) ).

When playing in a LAN world, the same port is chosen as minecrafts publish to lan port. This is completely unrelated to the server config as this is only applicable on dedicated servers.

Reply:
What would you say to a text like this?:
> The port number to use for the voice chat communication.
> Audio packets are always transmitted via the UDP protocol on the port number
> specified here, independently of other networking used for the game server.
> Set this to '-1' to use the same port number that is used by the Minecraft server.
> However, it is strongly recommended NOT to use the same port number because it
> is also used by default for the server query. Doing so may crash the server!

Yes, it's even longer, but now it clearly explains the causes and their effects. This way, more people may take the warning seriously rather than treating it like a vague "don't press the red button because I said so". Furthermore, mentioning "server query" will help guide the users, who have no choice but to use the same port number, that this is the thing that may be the source of their problems and that it may be a good idea to make some changes to the vanilla server properties.

As for readability, in my opinion, the problem is not with long descriptions because, for example, the (Craft)Bukkit plugin Shopkeepers has *very* long descriptions for its configuration options, yet this doesn't hinder the experience, and this actually helps quite a bit. Why? Because they are separated by blank lines (not all of them, unfortunately). And that's all that's needed.

Reply:
What would you say to a text like this?:
> The port number to use for the voice chat communication.
> Audio packets are always transmitted via the UDP protocol on the port number
> specified here, independently of other networking used for the game server.
> Set this to '-1' to use the same port number that is used by the Minecraft server.
> However, it is strongly recommended NOT to use the same port number because it
> is also used by default for the server query. Doing so may crash the server!

Yes, it's even longer, but now it clearly explains the causes and their effects. This way, more people may take the warning seriously rather than treating it like a vague "don't press the red button because I said so". Furthermore, mentioning "server query" will help guide the users, who have no choice but to use the same port number, that this is the thing that may be the source of their problems and that it may be a good idea to make some changes to the vanilla server properties.

As for readability, in my opinion, the problem is not with long descriptions because, for example, the (Craft)Bukkit plugin Shopkeepers has *very* long descriptions for its [configuration options](https://github.com/Shopkeepers/Shopkeepers/blob/master/modules/main/src/main/resources/config.yml), and this actually helps quite a bit. Why? Because they are separated by blank lines (not all of them, unfortunately). And that's all that's needed.

Reply:
What would you say to a text like this?:
> The port number to use for the voice chat communication.
> Audio packets are always transmitted via the UDP protocol on the port number
> specified here, independently of other networking used for the game server.
> Set this to '-1' to use the same port number that is used by the Minecraft server.
> However, it is strongly recommended NOT to use the same port number because it
> is also used by default for the server query. Doing so may crash the server!

Yes, it's even longer, but now it clearly explains the causes and their effects. This way, more people may take the warning seriously rather than treating it like a vague "don't press the red button because I said so". Furthermore, mentioning "server query" will help guide the users, who have no choice but to use the same port number, that this is the thing that may be the source of their problems and that it may be a good idea to make some changes to the vanilla server properties.

As for readability, in my opinion, the problem is not with long descriptions because, for example, the (Craft)Bukkit plugin Shopkeepers has *very* long descriptions for its [configuration options](https://github.com/Shopkeepers/Shopkeepers/blob/master/modules/main/src/main/resources/config.yml), and this actually helps quite a bit. Why? Because they are separated by blank lines (not all of them, unfortunately). And that’s all you need.

Reply:
Yeah I like that